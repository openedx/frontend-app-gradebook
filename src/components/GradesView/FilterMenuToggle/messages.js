import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  editFilters: {
    id: 'gradebook.GradesView.editFilterLabel',
    defaultMessage: 'Edit Filters',
    description: 'A labeled button in the Grades tab that opens/closes the Filters tab, allowing the grades to be filtered',
  },
});

export default messages;
