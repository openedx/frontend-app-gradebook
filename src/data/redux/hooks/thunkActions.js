import { StrictDict } from 'utils';
import thunkActions from 'data/thunkActions';
import { actionHook } from './utils';

const app = StrictDict({
  filterMenu: {
    useCloseMenu: actionHook(thunkActions.app.filterMenu.close),
    useHandleTransitionEnd: actionHook(thunkActions.app.filterMenu.handleTransitionEnd),
    useToggleMenu: actionHook(thunkActions.app.filterMenu.toggle),
  },
  useSetModalStateFromTable: actionHook(thunkActions.app.setModalStateFromTable),
});

const grades = StrictDict({
  useFetchGradesIfAssignmentGradeFiltersSet: actionHook(
    thunkActions.grades.fetchGradesIfAssignmentGradeFiltersSet,
  ),
  useFetchPrevNextGrades: actionHook(thunkActions.grades.fetchPrevNextGrades),
  useFetchGrades: actionHook(thunkActions.grades.fetchGrades),
  useSubmitImportGradesButtonData: actionHook(thunkActions.grades.submitImportGradesButtonData),
  useUpdateGrades: actionHook(thunkActions.grades.updateGrades),
});

export default StrictDict({
  app,
  grades,
});
