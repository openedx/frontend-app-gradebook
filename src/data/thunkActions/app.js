/* eslint-disable import/no-self-import */
import { StrictDict } from '@src/utils';

import actions from '@src/data/actions';
import selectors from '@src/data/selectors';
import { fetchGradeOverrideHistory } from './grades';
import { fetchRoles } from './roles';
import * as module from './app';

export const initialize = (courseId, urlQuery) => (dispatch) => {
  dispatch(actions.app.setCourseId(courseId));
  dispatch(actions.filters.initialize(urlQuery));
  dispatch(fetchRoles());
};

export const filterMenu = StrictDict({
  close: () => (dispatch, getState) => {
    if (selectors.app.filterMenu.open(getState())) {
      dispatch(module.filterMenu.toggle());
    }
  },
  handleTransitionEnd: (event) => (dispatch) => {
    if (event.currentTarget === event.target) {
      dispatch(actions.app.filterMenu.endTransition());
    }
  },
  toggle: () => (dispatch) => {
    dispatch(actions.app.filterMenu.startTransition());
    const toggleMenu = () => dispatch(actions.app.filterMenu.toggle());
    const animationCb = () => window.setTimeout(toggleMenu);
    window.requestAnimationFrame(animationCb);
  },
});

export const setModalStateFromTable = ({ userEntry, subsection }) => (
  (dispatch) => {
    dispatch(fetchGradeOverrideHistory(subsection.module_id, userEntry.user_id));
    dispatch(actions.app.setModalStateFromTable({ subsection, userEntry }));
  }
);

export default StrictDict({
  initialize,
  filterMenu,
  setModalStateFromTable,
});
