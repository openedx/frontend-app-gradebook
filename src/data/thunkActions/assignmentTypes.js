/* eslint-disable import/prefer-default-export */
import { StrictDict } from 'utils';

import lms from 'data/services/lms';
import actions from 'data/actions';

import { fetchBulkUpgradeHistory } from './grades';

const {
  assignmentTypes: { fetching, gotGradesFrozen },
  config: { gotBulkManagementConfig },
} = actions;

export const fetchAssignmentTypes = () => (
  (dispatch) => {
    dispatch(fetching.started());
    return lms.api.fetch.assignmentTypes()
      .then(response => response.data)
      .then((data) => {
        dispatch(fetching.received(Object.keys(data.assignment_types)));
        dispatch(gotGradesFrozen(data.grades_frozen));
        dispatch(gotBulkManagementConfig(data.can_see_bulk_management));
        if (data.can_see_bulk_management) {
          dispatch(fetchBulkUpgradeHistory());
        }
      })
      .catch(() => {
        dispatch(fetching.error());
      });
  }
);

export default StrictDict({ fetchAssignmentTypes });
