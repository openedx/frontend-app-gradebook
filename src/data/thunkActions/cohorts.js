/* eslint-disable import/prefer-default-export */
import { StrictDict } from 'utils';
import actions from 'data/actions';
import lms from 'data/services/lms';

export const fetchCohorts = () => (
  (dispatch) => {
    dispatch(actions.cohorts.fetching.started());
    return lms.api.fetch.cohorts()
      .then(({ data }) => {
        dispatch(actions.cohorts.fetching.received(data));
      })
      .catch(() => {
        dispatch(actions.cohorts.fetching.error());
      });
  }
);

export default StrictDict({ fetchCohorts });
