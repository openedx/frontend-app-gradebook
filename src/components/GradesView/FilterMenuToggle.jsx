import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { Button, Icon } from '@edx/paragon';
import { FormattedMessage } from '@edx/frontend-platform/i18n';

import thunkActions from 'data/thunkActions';

import messages from './FilterMenuToggle.messages';

/**
 * Controls for filtering the GradebookTable. Contains the "Edit Filters" button for opening the filter drawer
 * as well as the search box for searching by username/email.
 */
export const FilterMenuToggle = ({ toggleFilterDrawer }) => (
  <Button
    id="edit-filters-btn"
    className="btn-primary align-self-start"
    onClick={toggleFilterDrawer}
  >
    <Icon className="fa fa-filter" /> <FormattedMessage {...messages.editFilters} />
  </Button>
);

FilterMenuToggle.propTypes = {
  // From Redux
  toggleFilterDrawer: PropTypes.func.isRequired,
};

export const mapStateToProps = () => ({});

export const mapDispatchToProps = {
  toggleFilterDrawer: thunkActions.app.filterMenu.toggle,
};

export default connect(mapStateToProps, mapDispatchToProps)(FilterMenuToggle);
