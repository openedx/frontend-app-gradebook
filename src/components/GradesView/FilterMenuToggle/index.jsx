import React from 'react';

import { Button, Icon } from '@edx/paragon';
import { useIntl } from '@edx/frontend-platform/i18n';
import { FilterAlt } from '@edx/paragon/icons';

import { thunkActions } from 'data/redux/hooks';

import messages from './messages';

/**
 * Controls for filtering the GradebookTable. Contains the "Edit Filters" button for opening the filter drawer
 * as well as the search box for searching by username/email.
 */
export const FilterMenuToggle = () => {
  const toggleFilterMenu = thunkActions.app.filterMenu.useToggleMenu();
  const { formatMessage } = useIntl();
  return (
    <Button
      id="edit-filters-btn"
      className="btn-primary align-self-start"
      onClick={toggleFilterMenu}
    >
      <Icon src={FilterAlt} /> {formatMessage(messages.editFilters)}
    </Button>
  );
};

FilterMenuToggle.propTypes = {};

export default FilterMenuToggle;
