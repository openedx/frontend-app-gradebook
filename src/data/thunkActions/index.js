import { StrictDict } from '../../utils';
import app from './app';
import assignmentTypes from './assignmentTypes';
import cohorts from './cohorts';
import grades from './grades';
import roles from './roles';
import tracks from './tracks';

export default StrictDict({
  app,
  assignmentTypes,
  cohorts,
  grades,
  roles,
  tracks,
});
