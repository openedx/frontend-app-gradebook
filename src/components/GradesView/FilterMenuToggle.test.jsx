import React from 'react';
import { shallow } from 'enzyme';

import thunkActions from 'data/thunkActions';

import { FilterMenuToggle, mapDispatchToProps, mapStateToProps } from './FilterMenuToggle';

jest.mock('@edx/paragon', () => ({
  Button: () => 'Button',
  Icon: () => 'Icon',
}));
jest.mock('data/thunkActions', () => ({
  __esModule: true,
  default: {
    app: {
      filterMenu: { toggle: jest.fn() },
    },
  },
}));

describe('FilterMenuToggle component', () => {
  describe('snapshots', () => {
    test('basic snapshot', () => {
      const toggleFilterDrawer = jest.fn().mockName('this.props.toggleFilterDrawer');
      expect(shallow((
        <FilterMenuToggle {...{ toggleFilterDrawer }} />
      ))).toMatchSnapshot();
    });
  });
  describe('mapStateToProps', () => {
    test('does not connect any selectors', () => {
      expect(mapStateToProps({ test: 'state' })).toEqual({});
    });
  });
  describe('mapDispatchToProps', () => {
    test('toggleFilterDrawer from thunkActions.app.filterMenu.toggle', () => {
      expect(mapDispatchToProps.toggleFilterDrawer).toEqual(
        thunkActions.app.filterMenu.toggle,
      );
    });
  });
});
