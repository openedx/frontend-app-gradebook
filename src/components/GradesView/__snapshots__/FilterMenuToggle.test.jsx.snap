// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterMenuToggle component snapshots basic snapshot 1`] = `
<Button
  className="btn-primary align-self-start"
  id="edit-filters-btn"
  onClick={[MockFunction this.props.toggleFilterDrawer]}
>
  <Icon
    className="fa fa-filter"
  />
   
  <FormattedMessage
    defaultMessage="Edit Filters"
    description="Button text on Grades tab to open/close the Filters tab"
    id="gradebook.GradesView.editFilterLabel"
  />
</Button>
`;
